annotation-processing/src/main/java/com/exactpro/epfast/annotation/processing/FastProcessor.java

```diff
@@ -82,15 +87,60 @@ private void reportDuplicates(String fastTypeName, Collection<Element> elements)
     }
 
     private void buildCreatorClass(HashMap<String, Element> nameClassMap) throws IOException {
-        TypeName typeName = new TypeName("com.exactpro.epfast.annotation.internal.CreatorImpl");
+        TypeName provider = new TypeName("com.exactpro.epfast.annotation.internal.CreatorImpl");
+        TypeName service = new TypeName("com.exactpro.epfast.ICreator");
         // mustache files are in UTF-8 by default
         Mustache mustache = mustacheFactory.compile(
             "com/exactpro/epfast/annotation/processing/CreatorImpl.java.mustache");
-
-        JavaFileObject builderFile = processingEnv.getFiler().createSourceFile(typeName.toString());
+        Collection<Element> elements = nameClassMap.values();
+        addServiceEntry(service.toString(), provider.toString(), elements.toArray(new Element[0]));
```

Comment:
In this style `elements.toArray(new Element[elements.size()])` is preferable. But better to use stream API as it's more readable: `elements.stream().toArray(Element[]::new)`. BTW, In Java 11 we can finally just `elements.toArray(Element[]::new)`